osfoffline/polling_osf_manager/osf_query.py

```diff
@@ -242,34 +251,39 @@ def _delete_file_folder(self, remote_file_folder):
         resp.close()
 
     @asyncio.coroutine
-    def make_request(self, url, method=None, params=None, expects=None, get_json=False, timeout=180, data=None):
+    def make_request(self, url, method=None, params=None, expects=None, get_json=False, timeout=180, data=None, headers=None, local_file=None, locks=None):
         if method is None:
             method = 'GET'
 
         request = self.request_session.request(
             url=url,
             method=method.upper(),
+            headers=headers,
             params=params,
             data=data
         )
         try:
-            response = yield from asyncio.wait_for(request, timeout)
-        except (
-                aiohttp.errors.ClientTimeoutError, aiohttp.errors.ClientConnectionError,
+            if locks and local_file:
+                asyncio.async(check_for_changes(local_file, locks))
```

Comment:
As Matt said, the version of python we are using is python 3.4.3, it won't be too hard to change to python 3.4.4. But we need to set down a version to avoid future's change.